conans/client/tools/scm.py

```diff
@@ -220,3 +222,20 @@ def is_pristine(self):
     def get_revision(self):
         return self.run("info --show-item revision").strip()
 
+    def get_last_changed_revision(self):
+        return self.run("info --show-item last-changed-revision").strip()
+
+    def get_branch(self):
+        url = self.run("info --show-item relative-url").strip()
+        try:
+            pattern = "(tags|branches)/[^/]+|trunk"
+            branch = re.search(pattern, url)
+            
+            if branch is None:
+                return None
+            else:
+                # Replace non alphanumeric
+                branch = re.sub('[^0-9a-zA-Z]+', '_', branch[0])
+                return branch
```

Comment:
Why replace non-alphanumeric?

Reply:
I think this is for SemVer:
> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92.

Reply:
I just copied it from the Git branch function. May be we should add an option to the function, if it should be replace non alphanumeric or not?
`get_branch(replace_non_alphanumeric=True):`

Reply:
it is an anti-feature in the Git class, we should remove it from there as soon as possible

Reply:
https://github.com/conan-io/conan/pull/3496/files

Reply:
As soon as this PR gets into `develop` branch I updated my fork with those changes (needed to continue working here).

Reply:
Already merged. I think we can remove the branch-renaming stuff from here